PickAPic/PickAPic/ViewController.m

```diff
@@ -2,17 +2,85 @@
+//  Created by Varindra Hart on 9/17/15.
+//  Copyright © 2015 Varindra Hart. All rights reserved.
 //
 
 #import "ViewController.h"
+#import <QuartzCore/QuartzCore.h>
 
-@interface ViewController ()
+@interface ViewController ()<UIImagePickerControllerDelegate, UINavigationControllerDelegate, UIAlertViewDelegate>
+
+@property (nonatomic) IBOutlet UIImageView *imageView;
+@property (weak, nonatomic) IBOutlet UIButton *takeAPicButton;
+@property (weak, nonatomic) IBOutlet UIButton *useCameraRollButton;
 
 @end
 
 @implementation ViewController
 
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.takeAPicButton.layer.cornerRadius = 10;
+    self.useCameraRollButton.layer.cornerRadius = 10;
+    self.navigationItem.title = @"Pick A Pic";
+    
+    if (![UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        
+//        UIAlertController * alert = [UIAlertController alertControllerWithTitle:@"Warning" message:@"Attempting to run application on a non camera compliant device" preferredStyle:UIAlertControllerStyleAlert];
+//        [alert addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:nil]];
+//        [alert show];
+
+        UIAlertView * alert = [[UIAlertView alloc] initWithTitle:@"Warning" message:@"Attempting to run application on a non camera compliant device" delegate:self cancelButtonTitle:@"Cancel" otherButtonTitles:nil];
+        [alert show];
+    }
+    // Do any additional setup after loading the view, typically from a nib.
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+
+- (IBAction)takePic:(UIButton *)sender{
+    
+    UIImagePickerController *picker = [[UIImagePickerController alloc]init];
+    picker.delegate = self;
+    picker.allowsEditing = YES;
+    picker.sourceType = UIImagePickerControllerSourceTypeCamera;
+    
+    [self presentViewController:picker animated:YES completion:nil];
+    
+}
+
+- (IBAction)selectPicture:(UIButton *)sender{
+    UIImagePickerController *picker = [[UIImagePickerController alloc]init];
+    picker.delegate = self;
+    picker.allowsEditing = YES;
+    picker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    
+    [self presentViewController:picker animated:YES completion:nil];
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info {
+    
+    UIImage *chosenImage = info[UIImagePickerControllerEditedImage];
+    self.imageView.image = chosenImage;
+    
+    [picker dismissViewControllerAnimated:YES completion:NULL];
+    
+}
+
+- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker{
+    
+    [picker dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)buttonIndex{
+    
+    exit(0);
```

Comment:
This kills the app, after I dismiss the alertview. If I run this on a device without a camera (ios simulator), I can never reach your camera roll.